### DISCUSSION TITLE
How is the value for DeviceInfo_Id getting populated by the 1DS sdk?

### DISCUSSION THREAD

**Discussion by amarzavery** [microsoft/cpp_client_telemetry#1146] [Q&A]
We are building a Win32 desktop application and are using the 1DS CPP sdk for sending telemetry to the kusto cluser.
The column `DeviceInfo_Id` contain some guid. However, we don't understand how that value is populated. We tried to lookup the Device Object Id or the Intune Device Id in the Azure portal and the Intune portal (endpoint.microsoft.com). However, the guid is completely different.

It would be great if you could tell us how is the value of DeviceInfo_Id getting populated or what is being used to identify a device and generate an Id for it.

**Comment by lalitb** [comment]
It's the GUID of the first network adapter, as documented here - https://github.com/microsoft/cpp_client_telemetry/blob/e54fb7568afb7dc17b07ebb76181ae042449c505/lib/pal/desktop/WindowsDesktopDeviceInformationImpl.cpp#L47
